library/src/main/java/org/pollub/library/utils/ValidRoleSet.java

```diff
@@ -0,0 +1,16 @@
+package org.pollub.library.utils;
+
+import jakarta.validation.Constraint;
+import jakarta.validation.Payload;
+
+import java.lang.annotation.*;
+
+@Documented
+@Constraint(validatedBy = ValidRoleSetValidator.class)
+@Target({ElementType.FIELD, ElementType.PARAMETER})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface ValidRoleSet {
```

Comment:
I don't know what's going on here

Reply:
Whats the purpose of this

Reply:
Jakarta Bean Validation (and similar validation frameworks) don't have built-in support for validating custom enums, like role sets. So, you created this custom annotation to apply your own validation logic (inside the ValidRoleSetValidator class) for roles, ensuring that only valid roles are allowed in the system.

In summary, this annotation allows you to easily mark fields or parameters in your application that need role validation, and it delegates the actual validation logic to the ValidRoleSetValidator class.

Reply:
![image](https://github.com/user-attachments/assets/29922d26-5dab-48ed-b7c2-a436f33abd29)